cloudinit/stages.py

```diff
@@ -934,22 +953,7 @@ def _find_networking_config(self):
         )
 
     def _apply_netcfg_names(self, netcfg):
-        ncfg_instance_path = self.paths.get_ipath_cur("network_config")
-        network_link = self.paths.get_runpath("network_config")
-        if not self._network_already_configured():
-            if os.path.exists(ncfg_instance_path):
-                if netcfg != util.load_json(
-                    util.load_file(ncfg_instance_path)
-                ):
-                    atomic_helper.write_json(
-                        ncfg_instance_path, netcfg, mode=0o600
-                    )
-            else:
-                atomic_helper.write_json(
-                    ncfg_instance_path, netcfg, mode=0o600
-                )
-        if not os.path.islink(network_link):
-            util.sym_link(ncfg_instance_path, network_link)
+        self._netcfg = netcfg
```

Comment:
I don't think we should set this here this as we have always first called `self._find_networking_config()` before we call this.

Reply:
right, dropped. It's already being set on the instance.

Reply:
dropped in latest. We will perform this in apply_network_config with smarts to avoid writing the file unless `/var/lib/cloud/instance` exists.